peps/pep-9999.rst

```diff
@@ -0,0 +1,289 @@
+mechanism, satisfying all four conditions.
+
+Specification
+=============
+
+This PEP specifies two aspects: a set of project status markers,
+as well as their presentation in the standard HTML and JSON indices.
+
+Project status markers
+----------------------
+
+This PEP proposes the following project status markers.
+
+A project **MAY** have multiple status markers. When evaluating multiple status
+markers for index behavior, the index **MUST** use the strongest negative
+requirement. For example, if a project is both archived and quarantined, the
+index **MUST NOT** offer any distributions of the project for download.
+
+Indices **MAY** implement any subset of the status markers specified in this
+PEP, as applicable.
+
+``archived``
+~~~~~~~~~~~~
+
+Description: The project does not expect to be updated in the future.
+
+Index semantics:
+
+* The index hosting the project **MUST NOT** allow uploads of new distributions to
+  the project.
+* The index **MUST** offer existing distributions of the project for download.
+
+Installer semantics:
+
+* Installers **MAY** produce warnings about a project's archival.
+
+``quarantined``
+~~~~~~~~~~~~~~~
+
+Description: The project is considered generally unsafe for use, e.g. due to
+malware.
+
+Index semantics:
+
+* The index hosting the project **MUST NOT** allow uploads of new distributions to
+  the project.
+* The index **MUST NOT** offer any distributions of the project for download.
+
+Installer semantics:
+
+* Installers **MAY** produce warnings about a project's quarantine, although
+  doing so is effectively moot (as the index will not offer any distributions
+  for installation).
+
+``deprecated``
+~~~~~~~~~~~~~~
+
+Description: The project is considered obsolete, and may have been superseded
+by another project.
+
+Index semantics:
+
+* The index hosting the project **MUST** allow uploads of new distributions to
+  the project.
+* The index **MUST** offer existing distributions of the project for download.
+
+Installer semantics:
+
+* Installers **MAY** produce warnings about a project's deprecation.
+
+Status markers in the index APIs
+--------------------------------
+
+This PEP defines version 1.4 of the index APIs.
+
+HTML index
+~~~~~~~~~~
+
+The following changes are made to the
+:ref:`simple repository API <packaging:simple-repository-api-base>`:
+
+* The index **SHALL** define the ```pypi:repository-version`` as ``1.4``.
+* When a project has one or more statuses, the index **SHALL** add
+  an appropriate ``pypi:project-status`` meta tag for each.
+
+For example, the following would be a valid HTML index response for
+``sampleproject`` after is has been marked as both ``archived`` and
+``quarantined``:
+
+.. code-block:: html
+
+    <!DOCTYPE html>
+    <html>
+      <head>
+        <meta name="pypi:repository-version" content="1.4">
+        <meta name="pypi:project-status" content="archived">
+        <meta name="pypi:project-status" content="quarantined">
```

Comment:
Flagging: another option here would be to invent some syntax, e.g.:

```html
<meta name="pypi:project-status" content="archived,quarantined">
```

However, multiple `<meta>` tags with the same name are valid HTML, so I figured having a slightly more verbose representation was simpler to consume. But I'm open to alternatives.